slsDetectorPackage/slsDetectorSoftware/multiSlsDetector/multiSlsDetector.cpp

```diff
@@ -2899,10 +2899,12 @@ slsDetectorDefs::externalCommunicationMode multiSlsDetector::setExternalCommunic
 	externalCommunicationMode ret, ret1;
 	//(Dhanya) FIXME: why first detector or is it the master one?
 	if (detectors.size())
+    {
 		ret = detectors[0]->setExternalCommunicationMode(pol);
-	if (detectors[0]->getErrorMask())
-		setErrorMask(getErrorMask() | (1 << 0));
-
+        if (detectors[0]->getErrorMask())
+            setErrorMask(getErrorMask() | (1 << 0));
+    }
+    
```

Comment:
As tu renvoyé tes corrections à Dhanya pour qu'elle les récupère dans son code?

Reply:
Oui, j'ai envoyé un mail le 28 janvier 2020 et la modification devait être prise en compte dans le sdk.
Étant donné que le sdk à été réécrit en grande partie  (à partir de la version 5.0.0), je n'ai pas été en mesure de vérifier ce point car les méthodes impliquées n'existent plus.

-----------------------------------------------------------------------------------------------------------------------------------
Hello Dhanya,

I am working on several JIRA tickets for our Sls Eiger Lima Device and I would like to describe a problem found in the Sls Sdk.

I have a segmentation fault problem using the slsDetectorUsers::ReadConfigurationFile when the detector is not connected.
The problem comes from the setExternalCommunicationMode method. When the detectors container is empty, the detectors[0]->getErrorMask() call is causing the crash.

In the slsDetectorPackage\slsDetectorSoftware\multiSlsDetector\multiSlsDetector.cpp file :

slsDetectorDefs::externalCommunicationMode multiSlsDetector::setExternalCommunicationMode(
        externalCommunicationMode pol) {
    externalCommunicationMode ret, ret1;
    //(Dhanya) FIXME: why first detector or is it the master one?
    if (detectors.size())
        ret = detectors[0]->setExternalCommunicationMode(pol);
    if (detectors[0]->getErrorMask())
        setErrorMask(getErrorMask() | (1 << 0));

    for (unsigned int idet = 1; idet < detectors.size(); ++idet) {
        ret1 = detectors[idet]->setExternalCommunicationMode(pol);
        if (detectors[idet]->getErrorMask())
            setErrorMask(getErrorMask() | (1 << idet));
        if (ret != ret1)
            ret = GET_EXTERNAL_COMMUNICATION_MODE;
    }
    setMaster();
    setSynchronization();
    return ret;
}

My temporary correction for SOLEIL :

slsDetectorDefs::externalCommunicationMode multiSlsDetector::setExternalCommunicationMode(
        externalCommunicationMode pol) {
    externalCommunicationMode ret, ret1;
    //(Dhanya) FIXME: why first detector or is it the master one?
    if (detectors.size())
    {
        ret = detectors[0]->setExternalCommunicationMode(pol);
        if (detectors[0]->getErrorMask())
            setErrorMask(getErrorMask() | (1 << 0));
    }
   
    for (unsigned int idet = 1; idet < detectors.size(); ++idet) {
        ret1 = detectors[idet]->setExternalCommunicationMode(pol);
        if (detectors[idet]->getErrorMask())
            setErrorMask(getErrorMask() | (1 << idet));
        if (ret != ret1)
            ret = GET_EXTERNAL_COMMUNICATION_MODE;
    }
    setMaster();
    setSynchronization();
    return ret;
}

There is the same problem in the setExternalSignalFlags method.

Cheers,

Cédric